src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java

```diff
@@ -2,34 +2,54 @@
 
 import com.iridium.iridiumcore.utils.StringUtils;
 import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.database.Island;
 import com.iridium.iridiumskyblock.database.User;
 import com.iridium.iridiumskyblock.utils.PlayerUtils;
 import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 import org.bukkit.event.player.PlayerTeleportEvent;
 
+import java.util.Optional;
+
 public class PlayerTeleportListener implements Listener {
 
     @EventHandler(ignoreCancelled = true)
     public void onPlayerTeleport(PlayerTeleportEvent event) {
         if (event.getTo() == null) return;
-        User user = IridiumSkyblock.getInstance().getUserManager().getUser(event.getPlayer());
-        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getTo()).ifPresent(island -> {
-                    if (IridiumSkyblock.getInstance().getIslandManager().isBannedOnIsland(island, user)) {
-                        event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().youHaveBeenBanned
-                                .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
-                                .replace("%owner%", island.getOwner().getName())
-                                .replace("%name%", island.getName())
-                        ));
-                        event.setCancelled(true);
-                    } else {
-                        Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () ->
-                                        PlayerUtils.sendBorder(event.getPlayer(), island)
-                                , 1);
-                    }
+        Player player = event.getPlayer();
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        Optional<Island> islandOptional = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getTo());
+        
+        if (islandOptional.isPresent()) {
+            Island island = islandOptional.get();
+            if (!player.hasPermission("iridiumskyblock.bypassban") && IridiumSkyblock.getInstance().getIslandManager().isBannedOnIsland(island, user)) {
+                if (!user.isBypassing()) {
```

Comment:
Why does the player need to bypass too, now? I still don't get why we need that permission now, but both seems especially weird.

Reply:
Permission was present, i dont see what you want with this